src/vm/loaderallocator.cpp

```diff
@@ -2046,4 +2053,126 @@ BOOL LoaderAllocator::InsertComInteropData(MethodTable* pMT, InteropMethodTableD
 
 #endif // FEATURE_COMINTEROP
 
+
+void LoaderAllocator::AddUnmanagedImageToCache(LPCWSTR libraryName, HMODULE hMod)
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_ANY;
+        PRECONDITION(CheckPointer(libraryName));
+        INJECT_FAULT(COMPlusThrowOM(););
+    }
+    CONTRACTL_END;
+    if (libraryName)
```

Comment:
I don't understand why we have a contract that indicate non-null for `libraryName` and we protect against it? Seems like we only need one of these.

Also, should we have a contract check for `hMod` being null?

Reply:
Regarding all your questions on the contracts and various details, these functions were moved here from the AppDomain and I've only slightly touched them when necessary (to change the key in the cache from AssemblySpec* to LPCWSTR.) So these are the contracts and mostly code that we had there.

Reply:
@janvorli I figured that was the case. As long as we are touching these parts of the code, I think it is worth cleaning them up so they are consistent and coherent.